src/Middleware/HostFiltering/test/HostFilteringMiddlewareTests.cs

```diff
@@ -80,14 +80,14 @@ public async Task AllowsEmptyHost(bool allowed, int status)
                 {
                     app.Use((ctx, next) =>
                     {
-                        ctx.Request.Headers[HeaderNames.Host] = " ";
+                        ctx.Request.Headers[HeaderNames.Host] = "";
                         return next();
                     });
                     app.UseHostFiltering();
                     app.Run(c =>
                     {
-                        Assert.True(c.Request.Headers.TryGetValue(HeaderNames.Host, out var host));
-                        Assert.True(StringValues.Equals(" ", host));
+                        var host = c.Request.Headers[HeaderNames.Host];
+                        Assert.True(StringValues.Equals(new StringValues(""), host.ToString()));
```

Comment:
Does this maintain/test the distinction between there being no host header and there being an empty host header? The spec makes a distinction.

Reply:
Hmm... looking into it. Checking what the parser does.

Reply:
All looks ok (in Kestrel)

I think there is an issue with `Microsoft.AspNetCore.Http.HeaderDictionary`

https://github.com/aspnet/AspNetCore/blob/4649449206edae4992e09419334e736726fd2c8e/src/Http/Http/src/HeaderDictionary.cs#L77-L80

It should be testing an `IsNull` property (which doesn't exist) rather than the `IsNullOrEmpty`?

Reply:
StringValues doesn't even have an IsNull :-)
The issue you highlight would only be on the setter, it would be fine if the Store already contained the empty value. I think all of the servers implement their own request headers structure, and this isn't an issue for response headers. That just leaves TestServer?

Reply:
>  That just leaves TestServer?

Yes, but I assume its why the test was inserting `" "` rather than `""` as a value, as empty gets thrown away

Reply:
Ah, Kestrel does similar thing; but uses `.Count == 0` which I guess would be the equivalent of `.IsNull`

Reply:
Resolved